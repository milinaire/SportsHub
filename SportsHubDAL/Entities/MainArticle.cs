using SportsHubDAL.Interfaces;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace SportsHubDAL.Entities
{
    public class MainArticle : NoIdDBEntity
    {
        [Key]
        public int ArticleId { get; set; }
        public bool Show { get; set; }
        [ForeignKey("ArticleId")]
        public virtual Article Article { get; set; }
    }
}