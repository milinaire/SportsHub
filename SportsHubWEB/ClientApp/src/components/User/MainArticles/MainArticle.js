import React, {Fragment} from "react";
import style from './MainArticle.module.css'
import {Link} from "react-router-dom";

const MainArticle = (props) => {

  return (
    <Fragment>
      <div className={style.mainArticleWrapper}>
        <div className={style.mainArticleImg}>
          <div style={{
            backgroundImage: `url(${props.article.imageUri})`,
            width: '100%',
            height: '100%',
            backgroundRepeat: 'no-repeat',
            backgroundSize: '100% 100%',
          }}>
            <div className={style.mainArticleCategory}>
              {props.categories.length ? props.categories.find(c => c.id === props.article.categoryId).name : 'Loading...'}
            </div>
            <div className={style.mainArticleInfo}>
              Published / {new Date(props.article.datePublished).toLocaleDateString()}
              <div>{props.article.headline}</div>
              <div>{props.article.caption}</div>
              {props.link?<Link to={`${props.link}/${props.article.articleId}`}>More</Link>:'share'}
            </div>
            <div className={style.mainArticleButtons}>
              {props.buttons}
            </div>
          </div>
        </div>
      </div>
    </Fragment>
  );
}

export default MainArticle;
