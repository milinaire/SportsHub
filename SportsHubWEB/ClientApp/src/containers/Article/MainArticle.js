import React, {Component, Fragment} from "react";
import {withRouter} from "react-router-dom";

class MainArticle extends Component {
  constructor(props) {
    super(props);
    this.state = {
      Article: {}
    }
  }

  componentDidMount() {
    fetch(`/article/main/display`)
      .then(res => res.json())
      .then(
        (result) => {
          this.setState({
            Article: result.find(a => String(a.articleId) === String(this.props.match.params.article))
          });
          this.props.setMainArticles(result.filter(a => String(a.articleId) === String(this.props.match.params.article)), false, '')
        },
        (error) => {
          this.setState({
            error
          });
        }
      )
  }
  componentWillUnmount() {
    this.props.setMainArticles([])
  }

  render() {
    return (
      <Fragment>
        <div style={{padding: '50px'}}>
          <div className="text-a">
            <h3 className="text-lg-left">{this.state.Article.text}</h3>
          </div>
          <hr/>
        </div>
      </Fragment>
    )
  }
}

export default withRouter(MainArticle)