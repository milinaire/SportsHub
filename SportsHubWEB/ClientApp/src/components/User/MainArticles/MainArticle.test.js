import React from 'react';
import ReactDOM from 'react-dom';
import {MemoryRouter} from 'react-router-dom';

import renderer from 'react-test-renderer';
import MainArticle from "./MainArticle";

describe('my beverage', () => {
  const Props = {
    categories: [],
    article: {imageUri:'1'},
    setCurrentArticle: () => {
    }
  }
  test('renders without crashing', async () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <MemoryRouter>
        <MainArticle   {...Props}/>
      </MemoryRouter>, div);

  })
  test('', () => {
    const tree = renderer
      .create(<MainArticle  {...Props}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  })
})