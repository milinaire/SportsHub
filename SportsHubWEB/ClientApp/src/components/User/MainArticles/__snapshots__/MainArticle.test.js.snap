// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`my beverage  1`] = `
<div
  className="mainArticleWrapper"
>
  <div
    className="mainArticleImg"
  >
    <div
      style={
        Object {
          "backgroundImage": "url(1)",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "100% 100%",
          "height": "100%",
          "width": "100%",
        }
      }
    >
      <div
        className="mainArticleCategory"
      >
        Loading...
      </div>
      <div
        className="mainArticleInfo"
      >
        <div>
          <div
            className="mainArticleInfoDate"
          >
            Published / 
            Invalid Date
          </div>
          <div
            className="mainArticleInfoHeadline"
          />
          <div
            className="mainArticleInfoCaption"
          />
        </div>
        <div
          className="mainArticleInfoShare"
        >
          Share
        </div>
      </div>
      <div
        className="mainArticleButtons"
      />
    </div>
  </div>
</div>
`;
